sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/ResourceProviderData.Serialization.cs

```diff
@@ -61,7 +62,15 @@ public static ResourceProviderData DeserializeResourceProviderData(JsonElement e
                     List<ProviderResourceType> array = new List<ProviderResourceType>();
                     foreach (var item in property.Value.EnumerateArray())
                     {
-                        array.Add(ProviderResourceType.DeserializeProviderResourceType(item, options));
+                        if (options.TryGetProxy<ProviderResourceType>(out IJsonModel<ProviderResourceType> proxy))
```

Comment:
Should other model implementations in the test suite be updated to test for proxies in their implementations?

Reply:
hand writing this is a bit much to maintain we have a full set of autogenerated models in the generator since that is technically the code that owns the implementation.

Reply:
Will other tests fail in the future because the test client code is not up to date?